python/file_finder.py

```diff
@@ -0,0 +1,115 @@
+import os
+import subprocess
+import sys
+
+def fprint(msg, newline=True):
+    import sys
+    if newline:
+        print msg
+    else:
+        print msg,
+    sys.stdout.flush()
+
+#From: https://stackoverflow.com/questions/1883980/find-the-nth-occurrence-of-substring-in-a-string
+def find_nth(haystack, needle, n):
+    start = haystack.find(needle)
+    while start >= 0 and n > 1:
+        start = haystack.find(needle, start+len(needle))
+        n -= 1
+    return start
+
+def find_site(file_per_job, preferred_sites = None, prefer_us_sites = False, verbose = False):
+    file_and_site_per_job = {}
+    if verbose:
+        fprint("Finding the sites for each file ...", False)
+    for i, (job, file) in enumerate(file_per_job.iteritems()):
+        if file is None:
+            file_and_site_per_job[job] = (file,None,[None])
+        else:
+            cmd = "dasgoclient -query=\"site file=" + file + "\""
```

Comment:
would it be useful to print the das commands in verbose mode?

Reply:
Personally, I don't think so. It's not a complex command. For any given job the user could come up with the same command within a minute. If you feel strongly I could put it in.

Reply:
My reasoning here is that a user will report "it's not working" and be unable to provide more info. Having the subprocess commands printed explicitly can help debug.

Reply:
Done